### PR TITLE
adapt compiler to hledger

### PR DESCRIPTION
**Pull request by Konfekt**
This is a best effort to set up the compiler options for hledger as the current implementation and documentation does not seem adapted to it despite being documented

### CODE REVIEW

File: compiler/ledger.vim [hunk 1]
```diff
@@ -32,5 +32,8 @@ if !b:is_hledger
 	CompilerSet errorformat+=%-G%.%#
 	exe 'CompilerSet makeprg='.substitute(g:ledger_bin, ' ', '\\ ', 'g').'\ -f\ ' . substitute(shellescape(expand(g:ledger_main)), ' ', '\\ ', 'g') . '\ '.substitute(g:ledger_extra_options, ' ', '\\ ', 'g').'\ source\ ' . shellescape(expand(g:ledger_main))
 else
-	exe 'CompilerSet makeprg=('.substitute(g:ledger_bin, ' ', '\\ ', 'g').'\ -f\ ' . substitute(shellescape(expand(g:ledger_main)), ' ', '\\ ', 'g') . '\ print\ '.substitute(g:ledger_extra_options, ' ', '\\ ', 'g').'\ >\ /dev/null)'
+	let &l:makeprg='hledger check --strict' . substitute(g:ledger_extra_options, ' ', '\\ ', 'g') . '-f %:S'
```
Comment (alerque) [inline compiler/ledger.vim:35]: This looses the benefit of using `g:ledger_main` which I think is important. The ledger files I edit can't be validated properly without the declarations in my main entry point file that then includes the assorted ledgers I edit.

What problem were you initially trying to solve here?

Comment (Konfekt) [reply compiler/ledger.vim]: Yes, thank you for the quick review. I opened a pull request before finishing due to a typo. Should be fine now. It is best effort anyway. I had the impression that currently the compiler does not work at all for hledger despite being documented.

Comment (alerque) [reply compiler/ledger.vim]: It did work at one point. I use `hledger` but execute it with some concoctions of my own most of the time so I wouldn't have noticed if it stopped working. Our tests probably don't cover this (that would be nice).

Comment (Konfekt) [reply compiler/ledger.vim]: I checked with version 1.26.1. Are you sure it ever worked on Windows as the `(...) > /dev/null` construct seems specific to Unix ? As `compiler.vim` does not set any `&errorformat` for hledger, how to be sure that it was always correctly set?

Comment (alerque) [reply compiler/ledger.vim]: I don't know if it *ever* worked on Windows, no. Only that I remember fiddling with in on Linux a long time ago when some PR touched related code and it seemed to work.

**Review by alerque — APPROVED** [review]

**Pull request merged**